tests/test_dag.py

```diff
@@ -146,67 +146,66 @@ def test_end_nodes_dag(self):
 
 class DAGFailureTest(unittest.TestCase):
     def test_dag_failure(self):
-        with self.assertRaises(TypeError):
```

Comment:
So the behavior change here is the `compute` no longer raises the exception?

Reply:
@gsakkis can we make `compute` still raise the exception? Else now a user has no way to know there was a failure, per the test here `d.compute()` completes and returns without any signally to the user that it task graph actually failed.

Reply:
@Shelnutt2 `Dag.compute` is currently async, it just kicks off the dag computation without waiting for it to finish (and therefore know if an exception is raised). What we can do is raise the exception in `Dag.wait`. We can also make `Dag.compute` (optionally) synchronous by having it call `wait`; in fact that's what [DelayedBase.compute](https://github.com/TileDB-Inc/TileDB-Cloud-Py/pull/125/files#diff-2e84d094d407cf2aedc028330223fe426d6fef815326cf7f4d4cb445a25fa49cR41) does.

Reply:
I was intending `Delayed. compute()` and `Dag.wait()`. Doing it in `Dag.wait()` should be enough to propagate it up.

Reply:
Moved the reraise logic to `DAG.wait` at https://github.com/TileDB-Inc/TileDB-Cloud-Py/pull/125/commits/1f72ac9e83dfa2d849f7ccb9463c9d363ffe6450

Reply:
Thanks! I'll test and we can get this merged soon.